### ISSUE TITLE
StateError: Bad state: StreamSink is closed

### ISSUE THREAD

**Issue by eneserdemirci** [flutter/flutter#53254]
## command

flutter run -d chrome

## exception

StateError: Bad state: StreamSink is closed

```
#0      _StreamSinkImpl.add (dart:io/io_sink.dart:156:7)
#1      _Socket.add (dart:io-patch/socket_patch.dart:1759:38)
#2      _DetachedSocket.add (dart:_http/http_impl.dart:3066:13)
#3      _ByteAdapterSink.add (dart:convert/byte_conversion.dart:60:11)
#4      ByteConversionSinkBase.addSlice (dart:convert/byte_conversion.dart:45:5)
#5      _Utf8EncoderSink.addSlice (dart:convert/utf.dart:267:13)
#6      StringConversionSinkMixin.add (dart:convert/string_conversion.dart:165:5)
#7      SseConnection._setUpListener (package:sse/src/server/sse_handler.dart:73:15)
<asynchronous suspension>
#8      new SseConnection (package:sse/src/server/sse_handler.dart:53:15)
#9      SseHandler._createSseConnection.<anonymous closure> (package:sse/src/server/sse_handler.dart:161:26)
#10     _fromHttpRequest.onHijack.<anonymous closure> (package:shelf/shelf_io.dart:146:35)
#11     _rootRunUnary (dart:async/zone.dart:1155:38)
#12     _CustomZone.runUnary (dart:async/zone.dart:1048:19)
#13     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#14     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
#15     Future._propagateToListeners (dart:async/future_impl.dart:711:32)
#16     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#17     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:556:7)
#18     _rootRun (dart:async/zone.dart:1147:13)
#19     _CustomZone.run (dart:async/zone.dart:1040:19)
#20     _CustomZone.runGuarded (dart:async/zone.dart:942:7)
#21     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:982:23)
#22     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#23     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#24     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#25     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
```

## flutter doctor

```
[✓] Flutter (Channel master, v1.16.2-pre.49, on Microsoft Windows [Version 10.0.18363.720], locale tr-TR)
    • Flutter version 1.16.2-pre.49 at C:\src\flutter
    • Framework revision 10571920c0 (34 hours ago), 2020-03-23 18:47:50 -0700
    • Engine revision 2d42c74a34
    • Dart version 2.8.0 (build 2.8.0-dev.16.0 ba8baa46b4)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\theaz\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[✓] VS Code (version 1.43.2)
    • VS Code at C:\Users\theaz\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.8.1

[✓] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 80.0.3987.149
    • Web Server • web-server • web-javascript • Flutter Tools

• No issues found!
```

**Comment by VladyslavBondarenko** [comment]
Hi @eneserdemirci 
Can you reproduce it with `flutter run -d chrome --verbose` and attach logs here?

**Comment by eneserdemirci** [comment]
@VladyslavBondarenko


PS C:\Users\theaz\AndroidStudioProjects\vie> flutter run -d chrome --verbose
[  +82 ms] executing: [C:\src\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +145 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +53 ms] 10571920c0fba5a56c4142b38df82d1671ae9f0b
[   +3 ms] executing: [C:\src\flutter/] git describe --match v*.*.* --first-parent --long --tags
[ +133 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[   +1 ms] v1.16.2-7-g10571920c
[  +28 ms] executing: [C:\src\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +80 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[   +2 ms] origin/master
[   +1 ms] executing: [C:\src\flutter/] git ls-remote --get-url origin
[ +127 ms] Exit code 0 from: git ls-remote --get-url origin
[   +2 ms] https://github.com/flutter/flutter.git
[ +165 ms] executing: [C:\src\flutter/] git rev-parse --abbrev-ref HEAD
[  +56 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[   +2 ms] master
[  +97 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[  +24 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[  +80 ms] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[  +47 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[  +43 ms] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +27 ms] executing: C:\Users\theaz\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[  +93 ms] List of devices attached
[   +6 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[  +19 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +706 ms] Generating
C:\Users\theaz\AndroidStudioProjects\vie\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[ +473 ms] Launching lib\main.dart on Chrome in debug mode...
[  +43 ms] Updating assets
[+1146 ms] Syncing files to device Chrome...
[  +57 ms] Generating
C:\Users\theaz\AndroidStudioProjects\vie\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[ +222 ms] <- reset
[  +12 ms] C:\src\flutter\bin\cache\dart-sdk\bin\dart.exe
C:\src\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root C:\src\flutter\bin\cache\flutter_web_sdk/ 
--incremental --target=dartdevc --debugger-module-names -Ddart.developer.causal_async_stacks=true --output-dill
C:\Users\theaz\AppData\Local\Temp\flutter_tool.496471fd-6eae-11ea-a67b-d45ddf04023c\app.dill --libraries-spec
file:///C:/src/flutter/bin/cache/flutter_web_sdk/libraries.json --packages C:\Users\theaz\AndroidStudioProjects\vie\.packages
-Ddart.vm.profile=false -Ddart.vm.product=false
--bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions --enable-asserts --track-widget-creation --filesystem-root
C:\Users\theaz\AppData\Local\Temp\flutter_tools.496471fe-6eae-11ea-a67b-d45ddf04023c --filesystem-scheme org-dartlang-app
--initialize-from-dill build\cache.dill --platform file:///C:/src/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk.dill
[ +265 ms] <- compile org-dartlang-app:///web_entrypoint.dart
[+22498 ms] Syncing files to device Chrome... (completed in 22.883ms, longer than expected)
[   +8 ms] Synced 20.3MB.
[   +3 ms] <- accept
[+7190 ms] [CHROME]:
[   +5 ms] [CHROME]:DevTools listening on ws://127.0.0.1:64855/devtools/browser/cc34dff7-c27f-479d-85df-f609aea8728b
[+13791 ms] Debug service listening on ws://127.0.0.1:64911/7ymXomi8z8M=

[+1637 ms] Debug service listening on ws://127.0.0.1:64911/7ymXomi8z8M=
[   +5 ms] Warning: Flutter's support for web development is not stable yet and hasn't
[   +1 ms] been thoroughly tested in production environments.
[ +392 ms] For more information see https://flutter.dev/web
[  +38 ms]   To hot restart changes while running, press "r". To hot restart (and refresh the browser), press "R".
[   +5 ms] For a more detailed help message, press "h". To quit, press "q".

**Comment by VladyslavBondarenko** [comment]
@eneserdemirci 
I see it launched successfully. Can you reproduce the exception `StateError: Bad state: StreamSink is closed` or it happened only once?

**Comment by eneserdemirci** [comment]
@VladyslavBondarenko 

often this error happens ;
> Failed to establish connection with the application instance in Chrome.
> This can happen if the websocket connection used by the web tooling is unable to correctly establish a connection, for example due to a  
> firewall.

this happened once ; 

> StateError: Bad state: StreamSink is closed

**Comment by VladyslavBondarenko** [comment]
@eneserdemirci 

> Failed to establish connection with the application instance in Chrome.

Follow up on https://github.com/flutter/flutter/issues/49482 (and thumb it up)

> StateError: Bad state: StreamSink is closed

hardly this is actionable for now, closing.
Comment please if this happens again and you know how to reproduce it, and I'll reopen it (or create a new issue)
Thank you
